engine/src/main/java/org/terasology/world/block/family/BlockFamily.java

```diff
@@ -42,12 +43,12 @@
     /**
      * Get the block that is appropriate for placement in the given situation
      *
-     * @param location            The location where the block is going to be placed.
-     * @param attachmentSide      The side of the block which this block is being attached to, e.g. Top if the block is being placed on the ground
-     * @param direction           A secondary direction after the attachment side that determines the facing of the block.
+     * @param location          The location where the block is going to be placed
+     * @param attachmentSide    The side of the block which this block is being attached to, e.g. Top if the block is being placed on the ground
+     * @param viewingDirection  The player's viewing direction
      * @return The appropriate block
      */
-    Block getBlockForPlacement(Vector3i location, Side attachmentSide, Side direction);
+    Block getBlockForPlacement(Vector3i location, Side attachmentSide, Vector3fc viewingDirection);
```

Comment:
At least the method signature changes, so we can spot breaking families easily. I don't see an easy way to make this work in a backwards compatible way, so let's do this step! 

I'll check the Omega workspace for a full list of breaking block families 🙈

Reply:
Alright, I'll open a seperate PR for the API change.